### PR TITLE
Add initial structure for CR mapping integration test

### PR DESCRIPTION
**Pull request by tuommaki**
This change provides initial structure to introduce CR mapping integration test
for AzureConfig to CAPI CRs functionality.

### CODE REVIEW

File: integration/test/crmapping/crmapping_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			t.Log(tc.name)
+
+			client := newFakeClient()
+			ensureCRsExist(t, client, tc.inputFiles)
+
+			r := constructResource(t, client)
+
+			err := r.EnsureCreated(context.Background(), nil)
+
+			// TODO: Read expected CRs w/ client and compare to "golden" versions.
```
Comment (tuommaki) [inline integration/test/crmapping/crmapping_test.go:48]: In next PR I start the implementation of `capzcrs` functionality. Basic logic is required there in order to being able to then implement the gold file structure here. I'll provide both in the same PR then.

My idea here is to have set of input files that are "loaded" into fake client (i.e. the baseline before executing the handler) and then run the mapping once and then compare the resulting state with what is stored in "golden files".

This way it's easy to make thorough test coverage for CR fields while keeping maintenance work low when changing the mapping. One should then also see nicely the concrete diff of changes in end result CRs when changing the mapping logic.

File: integration/test/crmapping/crmapping_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	m := make(map[string]interface{})
+	err = yaml.Unmarshal(bs, m)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	switch m["kind"] {
+	case "AzureConfig":
+		obj = &providerv1alpha1.AzureConfig{}
+	default:
```
Comment (tuommaki) [inline integration/test/crmapping/crmapping_test.go:115]: I'll extend this switch later with relevant CAPI & CAPZ types.

File: integration/test/crmapping/testdata/simple_azureconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        port: 10250
+      networkSetup:
+        docker:
+          image: quay.io/giantswarm/k8s-setup-network-environment:1f4ffc52095ac368847ce3428ea99b257003d9b9
+        kubeProxy:
+          conntrackMaxPerCore: 0
+      ssh:
+        userList:
+        - name: foobar
+          publicKey: ssh-rsa foobarbaz== foobar
```
Comment (tuommaki) [inline integration/test/crmapping/testdata/simple_azureconfig.yaml:93]: This CR is from real live cluster but I redacted SSH keys out (even though they are public keys).

File: integration/test/crmapping/crmapping_test.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	"io/ioutil"
+	"path"
+	"strconv"
+	"testing"
+
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/giantswarm/operatorkit/resource"
+	"gopkg.in/yaml.v2"
```
Comment (axbarsan) [inline integration/test/crmapping/crmapping_test.go:14]: I've noticed that in most upstream projects they use `sigs.k8s.io/yaml` instead of `gopkg.in/yaml.v2`. Can we run into some edge cases where the behaviour would be different?

Comment (tuommaki) [reply integration/test/crmapping/crmapping_test.go:14]: I'm not aware of any. So far we have been always using `gopkg.in/yaml.v2` or `ghodss/yaml`. My quick grep over ~50 common repos found from my machine yields only few hits that are using k8s one and all rest using either one of aforementioned alternatives.

For consistency purposes I figure it would be a good thing to settle on one of these and change rest of them, but until anything breaks, it doesn't probably matter that much which one is used. While YAML itself is pretty complicated format with anchors & all, the used subset in k8s context is most of the time very simple so I don't expect to hit corner cases anytime soon. :slightly_smiling_face:

Comment (tuommaki) [reply integration/test/crmapping/crmapping_test.go]: Ok. I ended up having some struggles with YAML parsing in the end so among other changes I switched to k8s parser. Thanks for note! :+1:
